src/gui/effects/qgspainteffectwidget.cpp

```diff
@@ -117,13 +117,17 @@ QgsBlurWidget::QgsBlurWidget( QWidget *parent )
 {
   setupUi( this );
   connect( mBlurTypeCombo, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsBlurWidget::mBlurTypeCombo_currentIndexChanged );
-  connect( mBlurStrengthSpnBx, static_cast< void ( QSpinBox::* )( int ) >( &QSpinBox::valueChanged ), this, &QgsBlurWidget::mBlurStrengthSpnBx_valueChanged );
+  connect( mBlurStrengthSpnBx, static_cast< void ( QDoubleSpinBox::* )( double ) >( &QDoubleSpinBox::valueChanged ), this, &QgsBlurWidget::mBlurStrengthSpnBx_valueChanged );
+  connect( mBlurUnitWidget, &QgsUnitSelectionWidget::changed, this, &QgsBlurWidget::mBlurUnitWidget_changed );
   connect( mDrawModeComboBox, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsBlurWidget::mDrawModeComboBox_currentIndexChanged );
   connect( mBlendCmbBx, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsBlurWidget::mBlendCmbBx_currentIndexChanged );
 
-  mBlurTypeCombo->addItem( tr( "Stack blur (fast)" ), QgsBlurEffect::StackBlur );
+  mBlurTypeCombo->addItem( tr( "Stack blur (fast, no wider than 16px)" ), QgsBlurEffect::StackBlur );
```

Comment:
```suggestion
  mBlurTypeCombo->addItem( tr( "Stack blur (fast, limited size)" ), QgsBlurEffect::StackBlur );
```
?

Reply:
Actually -- do we even need to say this? Internally it'll fall back on guassian for large sizes anyway...

Reply:
@nyalldawson , I'm not sure we should do a silent gaussian fallback for the blur effect here as we are offering the stack vs gaussian method to begin with.